### ISSUE TITLE
[REQ] Convieniently only generate client classes

### ISSUE THREAD

**Issue by therealppa** [OpenAPITools/openapi-generator#5800]
I often generate client code from a spec file with the maven plugin, like this:

````
        <configuration>
          <inputSpec>src/main/oas/oas.yaml</inputSpec>

          <generatorName>java</generatorName>
          <library>native</library>

          <configOptions>
            <dateLibrary>java8</dateLibrary>
          </configOptions>

          <apiPackage>${project.groupId}.client</apiPackage>
          <modelPackage>${project.groupId}.client</modelPackage>
          <invokerPackage>${project.groupId}.client</invokerPackage>

          <generateApiTests>false</generateApiTests>
          <generateModelTests>false</generateModelTests>
          <generateModelDocumentation>false</generateModelDocumentation>
          <generateApiDocumentation>false</generateApiDocumentation>
          <supportingFilesToGenerate>
            ApiException.java,Pair.java,ApiClient.java
          </supportingFilesToGenerate>
        </configuration>
````

It would be nice to have a single config property that disables the generation of tests, documentation and also all the supporting files that are not needed for compilation.

It would also be nice to be able to specify the package name only once if you want everything in the same package. `packageName` does not do this.

**Comment by dragetd** [comment]
This seemes to be an ongoing wish back from the swagger times: https://github.com/swagger-api/swagger-codegen/issues/1737

The clunky defaults and lack of proper documentation on how to bend those into shape is frustrating when working with it. Your example actually helped me a lot right now. :S

**Comment by Woodham** [comment]
Honestly personally I'd go so far as to say this should really be the default for how the maven plugin is configured. It's certainly unexpected that a maven codegen plugin by default generates a whole separate maven project.

**Comment by therealppa** [comment]
So is there any chance this would ever get picked up?

**Comment by ppa-fyayc** [comment]
To my knowledge, nothing has changed in this area. Right?

**Comment by wing328** [comment]
I don't think anyone has started working on this yet. I've some ideas.

Would you like to contribute a PR or sponsor the work?

PM me via Slack if you want to discuss further

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g
